### PR TITLE
feat: expose metrics for all AWS API calls

### PR DESCRIPTION
**Pull request by threadproc**
### Description of your changes

This change will add a new Prometheus metric, `aws_api_calls_total`, exposed by controller-runtime that counts all requests made to the AWS APIs. This is useful when using expensive APIs like the Service Catalog API ($0.0007/request) for accounting purposes on accounts with multiple clusters.

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

This has been built and deployed to a test cluster where it was successfully observed to cause no errors when calling either v1 or v2 SDK methods, and successfully included the new metric on the `/metrics` endpoint.

### CODE REVIEW

File: pkg/controller/aws.go [hunk 1]
```diff
@@ -54,8 +54,8 @@ import (
 	"github.com/crossplane-contrib/provider-aws/pkg/controller/kinesis"
 	"github.com/crossplane-contrib/provider-aws/pkg/controller/kms"
 	"github.com/crossplane-contrib/provider-aws/pkg/controller/lambda"
-	"github.com/crossplane-contrib/provider-aws/pkg/controller/mq"
-	"github.com/crossplane-contrib/provider-aws/pkg/controller/mwaa"
+	mwaa "github.com/crossplane-contrib/provider-aws/pkg/controller/mq"
+	mq "github.com/crossplane-contrib/provider-aws/pkg/controller/mwaa"
```
Comment (MisterMX) [inline pkg/controller/aws.go:58]: Theses changes should not be made. A fix is provided with #1863.

File: pkg/controller/aws.go [hunk 2]
```diff
@@ -73,12 +73,15 @@ import (
...
 	"github.com/crossplane-contrib/provider-aws/pkg/controller/sqs"
 	"github.com/crossplane-contrib/provider-aws/pkg/controller/transfer"
+	"github.com/crossplane-contrib/provider-aws/pkg/utils/metrics"
 	"github.com/crossplane-contrib/provider-aws/pkg/utils/setup"
 )
 
 // Setup creates all AWS controllers with the supplied logger and adds them to
 // the supplied manager.
 func Setup(mgr ctrl.Manager, o controller.Options) error {
+	metrics.SetupMetrics(mgr, o)
```
Comment (MisterMX) [inline pkg/controller/aws.go:83]: I wonder if it would be better to call this in `cmd/provider/main.go` since this is some sort of "meta" feature that is not directly related to the controller setup.

**Review by MisterMX — CHANGES_REQUESTED** [review]
Thank you very much for your contribution, @threadproc! I think including AWS metric would a very meaningful enhancement.

I have made some small remarks that would need to be solved before merging. Can you take a look at them?

File: pkg/controller/aws.go (hunk 2 above)
Comment (threadproc) [reply pkg/controller/aws.go:83]: Agreed, I will make these changes

**Comment by threadproc** [comment]
@MisterMX I've made the requested changes and resolved some linting errors as well, thanks for the review!

**Comment by chlunde** [comment]
Great, very nice that you found a way to two this via hooks in the SDK!

File: cmd/provider/main.go [hunk 3]
```diff
@@ -123,6 +124,8 @@ func main() {
 		log.Info("Alpha feature enabled", "flag", features.EnableAlphaManagementPolicies)
 	}
 
+	metrics.SetupMetrics()
```
Comment (MisterMX) [inline cmd/provider/main.go:127]: To be consistent with the other setups, can we use `Register` instead of `MustRegister` and pass the error into `kingpin.FataIfError`? The result (a crash) would be the same, but it is more integrated into the control flow.

```suggestion
	kingpin.FatalIfError(metrics.SetupMetrics(), "Cannot setup AWS metrics hook")
```

File: pkg/utils/metrics/setup.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+var (
+	metricAWSAPICalls = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "aws_api_calls_total",
+		Help: "Number of API calls to the AWS API",
+	}, []string{"service", "operation", "api_version"})
+)
+
+// SetupMetrics will register the known Prometheus metrics with controller-runtime's metrics registry
+func SetupMetrics() {
+	k8smetrics.Registry.MustRegister(metricAWSAPICalls)
```
Comment (MisterMX) [inline pkg/utils/metrics/setup.go:18]:
```suggestion
func SetupMetrics() error {
	return k8smetrics.Registry.Register(metricAWSAPICalls)
```

**Review by MisterMX — CHANGES_REQUESTED** [review]

**Comment by threadproc** [comment]
@MisterMX I've implemented the requested changes and squashed them down again, thank you!

**Review by MisterMX — APPROVED** [review]
LGTM. Thank you very much for your contribution @threadproc!
